From 89483da629c05701ac2ff8a157a67e031e5e471f Mon Sep 17 00:00:00 2001
From: Mehdi Sabwat <mehdisabwat@gmail.com>
Date: Mon, 30 Dec 2019 13:55:23 +0100
Subject: [PATCH 1/1] contrib: add emscripten target

set toolchain variables in bootstrap and main.mak

Co-Author: Etienne Brateau <etienne.brateau@gmail.com>
---
 contrib/bootstrap    | 3 +++
 contrib/src/main.mak | 9 +++++++++
 2 files changed, 12 insertions(+)

diff --git a/contrib/bootstrap b/contrib/bootstrap
index 01a234e55c..b6224bae9d 100755
--- a/contrib/bootstrap
+++ b/contrib/bootstrap
@@ -348,6 +348,9 @@ case "${OS}" in
 	*nacl*)
 		add_make_enabled "HAVE_NACL"
 		;;
+	*emscripten*)
+	        add_make_enabled "HAVE_EMSCRIPTEN"
+		;;
 esac
 
 #
diff --git a/contrib/src/main.mak b/contrib/src/main.mak
index 5e5846de97..897020517e 100644
--- a/contrib/src/main.mak
+++ b/contrib/src/main.mak
@@ -133,6 +133,15 @@ EXTRA_CFLAGS += -fno-stack-check
 XCODE_FLAGS += OTHER_CFLAGS=-fno-stack-check
 endif
 
+ifdef HAVE_EMSCRIPTEN
+CC := emcc
+CXX := em++
+LD := emcc
+AR := emar
+RANLIB := emranlib
+CFLAGS="-pthread"
+endif
+
 ifdef HAVE_MACOSX
 EXTRA_CXXFLAGS += -stdlib=libc++
 ifeq ($(ARCH),x86_64)
-- 
2.24.1

